File: src/main/java/io/myosotisdev/utopianism/modules/stat/data/ParticleEffectData.java
package io.myosotisdev.utopianism.modules.stat.data;

import io.myosotisdev.utopianism.modules.effect.ParticleEffect;
import io.myosotisdev.utopianism.serialize.ParticleTagSerializer;
import io.myosotisdev.utopianism.serialize.Serializer;

public class ParticleEffectData
        implements ItemStatData<ParticleEffect>
{
    public static Serializer<ParticleEffect> Serializer = new ParticleTagSerializer();

    @Override
    public ParticleEffect getData()
    {
        return null;
    }
}

File: src/main/java/io/myosotisdev/utopianism/modules/stat/data/ParticleEffectData.kt
package io.myosotisdev.utopianism.modules.stat.data

import io.myosotisdev.utopianism.modules.effect.ParticleEffect
import io.myosotisdev.utopianism.serialize.ParticleTagSerializer
import io.myosotisdev.utopianism.serialize.Serializer

class ParticleEffectData : ItemStatData<ParticleEffect>
{
    override var value: ParticleEffect = ParticleEffect()

    companion object
    {
        val Serializer: Serializer<ParticleEffect> = ParticleTagSerializer()
    }
}